.github/workflows/generate.yml

```diff
@@ -36,19 +46,35 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
-
+                
       - name: Run config generator script
         env:
           NETWORK: ${{ github.event.inputs.network }}
           NAMESPACE: ${{ github.event.inputs.namespace }}
           ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+        if: ${{ github.event.inputs.timestampMethod == 'timestamp' }}
         run: |
           echo "Generating config with genesis timestamp ${{ github.event.inputs.genesisTime }} for network ${{ github.event.inputs.network }}-${{ github.event.inputs.namespace }}"
           chmod +x ./updateGenesisTime.sh ./genesis-state-gen-dev
           ./updateGenesisTime.sh ${{ github.event.inputs.network }} ${{ github.event.inputs.namespace }} ${{ github.event.inputs.genesisTime }}
           export SHA256_SUM=$(sha256sum ./$NETWORK/$NAMESPACE/network-config.yaml)
           echo "::set-env name=SHA256_SUM::$SHA256_SUM"
-
+          
+      - name: Run config generator script with delay
+        env:
```

Comment:
If we are duplicating `env` I think we can move them to the top of the workflow, because scope right now is wider than one workflow step (we are duplicating in two steps the same behavior).